src/Components/ServiceScene/Breakdowns/LineFilterInput.tsx

```diff
@@ -1,41 +1,96 @@
 import { css } from '@emotion/css';
-import { Icon, IconButton, Input, useStyles2 } from '@grafana/ui';
-import React, { HTMLProps } from 'react';
+import { Icon, IconButton, Input, Tooltip, useStyles2 } from '@grafana/ui';
+import React, { ChangeEvent, HTMLProps, useState } from 'react';
 import { GrafanaTheme2 } from '@grafana/data';
+import { narrowErrorMessage } from '../../../services/narrowing';
 
-interface Props extends Omit<HTMLProps<HTMLInputElement>, 'width' | 'prefix'> {
+interface Props extends Omit<HTMLProps<HTMLInputElement>, 'width' | 'prefix' | 'invalid' | 'onInvalid'> {
   onClear?: () => void;
   suffix?: React.ReactNode;
   prefix?: React.ReactNode;
   width?: number;
+  regex: boolean;
+  value: string;
 }
 
-export const LineFilterInput = ({ value, onChange, placeholder, onClear, suffix, width, ...rest }: Props) => {
+let re2JS: typeof import('re2js').RE2JS | undefined | null = undefined;
+
+export const LineFilterInput = ({ value, onChange, placeholder, onClear, suffix, width, regex, ...rest }: Props) => {
   const styles = useStyles2(getStyles);
-  return (
-    <Input
-      rows={2}
-      width={width}
-      value={value}
-      onChange={onChange}
-      suffix={
-        <span className={styles.suffixWrapper}>
-          {onClear && value ? (
-            <IconButton
-              aria-label={'Clear line filter'}
-              tooltip={'Clear line filter'}
-              onClick={onClear}
-              name="times"
-              className={styles.clearIcon}
-            />
-          ) : undefined}
-          {suffix && suffix}
-        </span>
+  const [invalid, setInvalid] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const load = async () => {
+    re2JS = null;
+    re2JS = (await import('re2js')).RE2JS;
+  };
+
+  const initValidation = (value: string) => {
+    if (re2JS === undefined && regex) {
+      load().then(() => validate(value));
+    } else if (regex && re2JS) {
+      validate(value);
+    }
+  };
```

Comment:
initValidation  is another candidate for useCallback  or perhaps useEffect to run on value change